pkg/tfbridge/schema_test.go

```diff
@@ -2172,6 +2172,58 @@ func TestRefreshExtractInputsFromOutputsMaxItemsOne(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestRefreshExtractInputsFromOutputsListOfObjects(t *testing.T) {
+	t.Parallel()
+
+	ruleSetProps := resource.PropertyMap{
+		"attachedDisks": resource.NewArrayProperty([]resource.PropertyValue{
+			resource.NewObjectProperty(resource.PropertyMap{
+				"name":   resource.NewStringProperty("name1"),
+				"key256": resource.NewNullProperty(),
+			}),
+		}),
+	}
+
+	ruleSetSchema := func() shim.SchemaMap {
+		blockList := func(elem schema.SchemaMap) shim.Schema {
+			s := schema.Schema{
+				Type:     shim.TypeList,
+				Optional: true,
+				Elem: (&schema.Resource{
+					Schema: elem,
+				}).Shim(),
+			}
+			return s.Shim()
+		}
+
+		return schema.SchemaMap{
+			"attachedDisks": blockList(schema.SchemaMap{
+				"name":   (&schema.Schema{Type: shim.TypeString, Optional: true}).Shim(),
+				"key256": (&schema.Schema{Type: shim.TypeString, Computed: true}).Shim(),
+			}),
+		}
+	}
+
+	ruleSetPs := func() map[string]*SchemaInfo {
+		return map[string]*SchemaInfo{
+			"attachedDisks": {
+				Fields: map[string]*SchemaInfo{
+					"name":   {},
```

Comment:
In `ruleSetProps` `name` is at the path `attachedDisks[0].name`. `ruleSetSchema` also defines `name` with `attachedDisks: list({"name", "key256"})`.

`ruleSetPs` defines name at `["attachedDisks"].Fields["name"]` (without the list index). I think it should be `["attachedDisks"].Elem.Fields["name"]`.

If you want to make sure that a schema override propagated, I would use a `Name:` override for `"name"`: `"name": {Name: "renamed"}`.